source/fuzz/transformation_split_loop.cpp

```diff
@@ -0,0 +1,693 @@
+        } else if (user->opcode() == SpvOpBranchConditional) {
+          auto instr_to_insert_before = user;
+          if (user_block->IsLoopHeader()) {
+            instr_to_insert_before = user->PreviousNode();
+          }
+          if (operand == 2) {
+            // If we have an instruction of form: "OpBranchConditional %cond
+            // %other %merge" then set |run_second| to the value of the boolean
+            // condition %cond
+            instr_to_insert_before->InsertBefore(MakeUnique<opt::Instruction>(
+                opt::Instruction(ir_context, SpvOpStore, 0, 0,
+                                 opt::Instruction::OperandList(
+                                     {{SPV_OPERAND_TYPE_ID,
+                                       {message_.variable_run_second_id()}},
+                                      {SPV_OPERAND_TYPE_ID,
+                                       {user->GetSingleWordInOperand(0)}}}))));
+          } else if (operand == 1) {
+            // If we have an instruction of form: "OpBranchConditional %cond
+            // %merge %other" then set |run_second| to the negation of the value
+            // of the boolean condition %cond.
+            auto result_id = logical_not_fresh_ids.back();
+            logical_not_fresh_ids.pop_back();
+            instr_to_insert_before->InsertBefore(
+                MakeUnique<opt::Instruction>(opt::Instruction(
+                    ir_context, SpvOpLogicalNot, bool_type_id, result_id,
+                    opt::Instruction::OperandList(
+                        {{SPV_OPERAND_TYPE_ID,
+                          {user->GetSingleWordInOperand(0)}}}))));
+            instr_to_insert_before->InsertBefore(MakeUnique<opt::Instruction>(
+                opt::Instruction(ir_context, SpvOpStore, 0, 0,
+                                 opt::Instruction::OperandList(
+                                     {{SPV_OPERAND_TYPE_ID,
+                                       {message_.variable_run_second_id()}},
+                                      {SPV_OPERAND_TYPE_ID, {result_id}}}))));
+          }
+        }
+      });
+
+  // In the terminator of |loop_header| take first branch which is not a merge
+  // block and set it to |new_body_entry|, which will be the first block of the
+  // body of the loop.
+  auto loop_header_terminator = loop_header_block->terminator();
+  uint32_t next_not_merge_id;
+  uint32_t next_not_merge_pos;
+  if (loop_header_terminator->opcode() == SpvOpBranch) {
+    next_not_merge_id = loop_header_terminator->GetSingleWordInOperand(0);
+    next_not_merge_pos = 0;
+  } else /*SpvOpBranchConditional*/ {
+    uint32_t first_id = loop_header_terminator->GetSingleWordInOperand(1);
+    uint32_t second_id = loop_header_terminator->GetSingleWordInOperand(2);
+    if (first_id != merge_block->id()) {
+      next_not_merge_id = first_id;
+      next_not_merge_pos = 1;
+    } else {
+      next_not_merge_id = second_id;
+      next_not_merge_pos = 2;
+    }
+  }
```

Comment:
Do we need to set `run_second` to false here if we branch directly from the heaeder block to the merge block?

Reply:
This is a special case of a branch to the merge block. All cases of branches (and conditional branches) to the merge block are handled later in the code to set run_second.